cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -81,4 +127,205 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    File uuidFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid);
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    if (!uuidFile.exists()) {
+      responder.sendString(HttpResponseStatus.OK, String.format("No such uuid %s in Support Bundle.", uuid));
+      return;
+    }
+    if (folderName == null || folderName.length() == 0 || dataFileName == null || dataFileName.length() == 0) {
+      File[] pipelineFiles =
+        uuidFile.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+      JsonArray pipelineArray = new JsonArray();
+      if (pipelineFiles != null && pipelineFiles.length > 0) {
+        for (File pipelineFile : pipelineFiles) {
+          if (folderName != null && folderName.length() > 0) {
+            if (pipelineFile.getName().equals(folderName)) {
+              JsonObject singlePipelineJson = bundleService.getFilesName(pipelineFile);
+              responder.sendString(HttpResponseStatus.OK, GSON.toJson(singlePipelineJson));
+              return;
+            }
+          } else {
+            pipelineArray.add(bundleService.getFilesName(pipelineFile));
+          }
+        }
+      }
+      responder.sendString(HttpResponseStatus.OK, GSON.toJson(pipelineArray));
+      return;
+    }
+    File folderDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid + "/" + folderName);
+    File[] dataFiles =
+      folderDirectory.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+    if (dataFiles != null && dataFiles.length > 0) {
+      for (File dataFile : dataFiles) {
+        if (dataFile.getName().startsWith(dataFileName)) {
+          BufferedReader br = new BufferedReader(new FileReader(dataFile));
+          StringBuilder sb = new StringBuilder();
+          String line = br.readLine();
+          while (line != null) {
+            sb.append(line).append("\n");
+            line = br.readLine();
+          }
+          responder.sendString(HttpResponseStatus.OK, sb.toString());
+          return;
+        }
+      }
+    }
+    responder.sendString(HttpResponseStatus.NOT_FOUND,
+                         String.format("uuid: %s with pipeline file name: %s and data file name: %s not found", uuid,
+                                       folderName, dataFileName));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
+  public void appsExport(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.CREATE);
```

Comment:
This should probably use enforce instead of enforceOnParent and enforce upon the specific SupportBundleId instead of on the parent. This should also use `StandardPermission.GET` as you are reading a support bundle instead of creating one.